dbos/workflow.go

```diff
@@ -162,12 +178,33 @@ type workflowHandle[R any] struct {
 	outcomeChan chan workflowOutcome[R]
 }
 
-func (h *workflowHandle[R]) GetResult() (R, error) {
-	outcome, ok := <-h.outcomeChan // Blocking read
-	if !ok {
-		// Return an error if the channel was closed. In normal operations this would happen if GetResul() is called twice on a handler. The first call should get the buffered result, the second call find zero values (channel is empty and closed).
-		return *new(R), errors.New("workflow result channel is already closed. Did you call GetResult() twice on the same workflow handle?")
+func (h *workflowHandle[R]) GetResult(opts ...GetResultOption) (R, error) {
+	options := &getResultOptions{}
+	for _, opt := range opts {
+		opt(options)
 	}
+
+	var timeoutChan <-chan time.Time
+	if options.timeout > 0 {
+		timeoutChan = time.After(options.timeout)
+	}
+
+	select {
+	case outcome, ok := <-h.outcomeChan:
+		if !ok {
+			// Return error if channel closed (happens when GetResult() called twice)
+			return *new(R), errors.New("workflow result channel is already closed. Did you call GetResult() twice on the same workflow handle?")
+		}
+		return h.processOutcome(outcome)
+	case <-h.dbosContext.Done():
+		return *new(R), h.dbosContext.Err()
```

Comment:
@maxdml I'm unable to get to this code path.

Tried shutting down DBOS server, but it get DB error from `h.outcomeChan` instead.

Let me know your thoughts

Reply:
For this case, should it be `context.Canceled` instead?